pkg/webhook/webhook.go

```diff
@@ -429,6 +429,58 @@ func createVolPatch(patch []jsonPatchOperation, hugepageResourceList []hugepageR
 	return patch
 }
 
+func addEnvVar(patch []jsonPatchOperation, containerIndex int, firstElement bool,
+	envName string, envVal string) []jsonPatchOperation {
+
+	env := corev1.EnvVar{
+		Name:  envName,
+		Value: envVal,
+	}
+
+	if firstElement {
+		patch = append(patch, jsonPatchOperation{
+			Operation: "add",
+			Path:      "/spec/containers/" + fmt.Sprintf("%d", containerIndex) + "/env",
+			Value:     []corev1.EnvVar{env},
+		})
+	} else {
+		patch = append(patch, jsonPatchOperation{
+			Operation: "add",
+			Path:      "/spec/containers/" + fmt.Sprintf("%d", containerIndex) + "/env/-",
+			Value:     env,
+		})
+	}
+
+	return patch
+}
+
+func createEnvPatch(patch []jsonPatchOperation, container *corev1.Container,
+	containerIndex int, envName string, envVal string) []jsonPatchOperation {
+
+	// Determine if requested ENV already exists
+	found := false
+	firstElement := false
+	if len(container.Env) != 0 {
+		for _, env := range container.Env {
+			if env.Name == envName {
+				found = true
+				if env.Value != envVal {
+					glog.Infof("Error, adding env %s, name existed but value different: %s != %s",
```

Comment:
I see there are three format specifiers here (%s x 3) and two values, is this ok? Also, I really like enclosing the specifiers in single quotes as well but not necessary here.
Nit: Also, do you think this should be a warning instead of info?

Reply:
@martinkennelly  Fixed. Thanks.

Reply:
I was thinking more about using glog.Warningf. glog.info could get filtered out.

Reply:
I did change to Warningf - Look at my last commit.